### PR TITLE
Support memory in react agent

### PR DESCRIPTION
**Pull request by Disiok**
# Description

Support memory in react agent

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

### CODE REVIEW

File: llama_index/agent/react/base.py [hunk 1]
```diff
@@ -70,12 +73,13 @@ def from_tools(
     ) -> "ReActAgent":
         tools = tools or []
         chat_history = chat_history or []
-        llm = llm or OpenAI(model="gpt-3.5-turbo-0613")
+        memory = memory or memory_cls.from_defaults(chat_history)
```
Comment (logan-markewich) [inline llama_index/agent/react/base.py:76]: Nit: should probably use the kwarg here

**Review by logan-markewich — APPROVED** [review]
Looks good to me! Works with from_tools, adds the user message and agent response

File: llama_index/agent/react/base.py (hunk 1 above)
Comment (Disiok) [reply llama_index/agent/react/base.py:76]: just noticed that we are also not passing llm as kwarg into the `from_defaults`, which means that we are not properly setting the max size of the memory according to the LLM.

Slipping in this fix for other places where we instantiate a memory object as well.

Comment (logan-markewich) [reply llama_index/agent/react/base.py]: Good catch!

**Pull request merged**